### PR TITLE
ISSUE-4459: Add support for alter table rename statement

### LINKED ISSUE #4459: ALTER TABLE [ IF EXISTS ] <name> RENAME TO <new_table_name>
**Summary**

After https://github.com/datafuselabs/databend/pull/4425 merged, it's time to add:
```
ALTER TABLE [ IF EXISTS ] <name> RENAME TO <new_table_name>
```

### PR DESCRIPTION
**Pull request by kevinw66**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add support for alter table rename statement

## Changelog

- New Feature

## Related Issues

Fixes #4459 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: query/src/sql/statements/statement_alter_table.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    pub table_name: ObjectName,
+    pub action: AlterTableAction,
+
+    // for AlterTableAction
+    pub new_table_name: ObjectName,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Copy, Debug, PartialEq)]
+pub enum AlterTableAction {
+    RenameTo,
```
Comment (Junnplus) [inline query/src/sql/statements/statement_alter_table.rs:41]:
```suggestion
    RenameTable(ObjectName),
```

We should put action and data together.

File: query/src/sql/statements/statement_alter_table.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+impl DfAlterTable {
+    fn resolve_table(
+        &self,
+        ctx: Arc<QueryContext>,
+        table_name: &ObjectName,
+    ) -> Result<(String, String)> {
+        let idents = &table_name.0;
+        match idents.len() {
+            0 => Err(ErrorCode::SyntaxException("Rename table name is empty")),
```
Comment (Junnplus) [inline query/src/sql/statements/statement_alter_table.rs:80]:
```suggestion
            0 => Err(ErrorCode::SyntaxException("Alter table name is empty")),
```

File: query/src/sql/statements/statement_alter_table.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            1 => Ok((ctx.get_current_database(), idents[0].value.clone())),
+            2 => Ok((idents[0].value.clone(), idents[1].value.clone())),
+            _ => Err(ErrorCode::SyntaxException(
+                "Rename table name must be [`db`].`table`",
```
Comment (Junnplus) [inline query/src/sql/statements/statement_alter_table.rs:84]:
```suggestion
                "Alter table name must be [`db`].`table`",
```

File: query/src/sql/statements/statement_alter_table.rs (hunk 2 above)
Comment (kevinw66) [reply query/src/sql/statements/statement_alter_table.rs:80]: oh, sorry I forgot this

**Comment by kevinw66** [comment]
@BohuTANG I saw stateless tests failed, but I can run in my local machine.
why?
image

**Comment by BohuTANG** [comment]
```
[ FAIL ] - result differs with:
--- /Users/runner/work/databend/databend/tests/suites/0_stateless/05_ddl/05_0003_ddl_alter_table.result	2022-03-22 05:20:32.000000000 +0000
+++ /Users/runner/work/databend/databend/tests/suites/0_stateless/05_ddl/05_0003_ddl_alter_table.stdout	2022-03-22 05:26:41.000000000 +0000
@@ -1,2 +1,3 @@
 1
+ERROR 1105 (HY000) at line 10: Code: 1025, displayText = Unknown table 't0'.
```

I think `t0` is a common name, some others test maybe use that? We can change it to a new unique name.

**Pull request merged**